java/vector/src/main/java/org/apache/arrow/vector/table/README.md

```diff
@@ -0,0 +1,384 @@
+
+```java
+FieldReader nameReader = table.getReader("user_name");
+```
+
+## Table API: Row operations
+
+Row-based access is supported using a Row object. Row provides *get()* methods by both vector name and vector position, but no *set()* operations. It is important to recognize that it's NOT a reified row, but rather operates like a cursor where the data from numerous logical rows in the Table can be viewed (one row at a time) using the same Row instance. See "Getting around" below for information about how to navigate through the table.
+
+**Note**: A mutable row implementation is expected with the release of MutableTable, which will support both mutable and immutable Rows.
+
+### Getting a row
+
+Calling `immutableRow()` on any table instance returns a row supporting read operations.
+
+```java
+Row r = table.immutableRow(); 
+```
+
+### Getting around
+
+Since rows are iterable, you can traverse a table using a standard while loop:
+
+```java
+Row r = table.immutableRow();
+    while (c.hasNext()) {
+    c.next();
+    // do something useful here
+    }
+```
+
+Table implements `Iterable<Row>` so you can access rows directly from Table in an enhanced *for* loop:
+
+```java
+for (Row row: table) {
+    int age = row.getInt("age");
+    boolean nameIsNull = row.isNull("name");
+    ...
+    }
+```
+
+Finally, while rows are usually iterated in the order of the underlying data vectors, but they are also positionable using the `Row#setPosition()` method, so you can skip to a specific row. Row numbers are 0-based.
+
+```java
+Row r = table.immutableRow();
+    int age101 = c.setPosition(101); // change position directly to 101
+```
+
+Any changes to position are of course applied to all the columns in the table.
+
+Note that you must call `next()`, or `setPosition()` before accessing values via a row. Failure to do so results in a runtime exception.
+
+### Read operations using rows
+
+Methods are available for getting values by vector name and vector index, where index is the 0-based position of the vector in the table. For example, assuming 'age' is the 13th vector in 'table', the following two gets are equivalent:
+
+```java
+Row r = table.immutableRow();
+    c.next(); // position the row at the first value
+    int age1 = c.get("age"); // gets the value of vector named 'age' in the table at row 0
+    int age2 = c.get(12);    // gets the value of the 13th vecto in the table at row 0
+```
+
+You can also get value using a NullableHolder. For example:
+
+```Java
+NullableIntHolder holder = new NullableIntHolder();
+    int b = row.getInt("age", holder);
+```
+
+This can be used to retrieve values without creating a new Object for each.
+
+In addition to getting values, you can check if a value is null using `isNull()` and you can get the current row number:
+
+```java
+boolean name0isNull = row.isNull("name");
+    int row = row.getRowNumber(); 
```

Comment:
It may be good to use `long` instead of `int` if we do anticipate adding ChunkedArray support in the future

Reply:
Will do

Reply:
fixed here and elsewhere in the doc